File: 2442-count-number-of-distinct-integers-after-reverse-operations/2442-count-number-of-distinct-integers-after-reverse-operations.kt
class Solution {
    fun countDistinctIntegers(nums: IntArray): Int {
        val set = HashSet<Int>()

        nums.forEach { num ->
            set.add(num)
            val revInt = num.toString().reversed().toInt()
            set.add(revInt)
        }

        return set.size
    }
}

File: 2442-count-number-of-distinct-integers-after-reverse-operations/2442-count-number-of-distinct-integers-after-reverse-operations.swift
class Solution {
    
    private func reverseNumber(_ num: Int) -> Int {
        return Int(String(String(num).reversed()))!
    }
    
    func countDistinctIntegers(_ nums: [Int]) -> Int {
        var set: Set<Int> = []
        
        nums.forEach { num in
            set.insert(num)
            set.insert(reverseNumber(num))
        }
        
        return set.count
    }
}
